### PR TITLE
Automated perf tests

### PR DESCRIPTION
**Pull request by ayumi**
#10115

Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Test Plan:

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -37,3 +38,5 @@ addons:
     packages:
       - xvfb
       - g++-4.8
+      - libgnome-keyring-dev
```
Comment (NejcZdovc) [inline .travis.yml]: We just remove this dependency in #10514. Do we need it back?

Comment (ayumi) [inline .travis.yml]: oops, was a mis-rebase. good catch!

File: .travis.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 language: node_js
 node_js:
-  - "7"
+  - "8"
```
Comment (luixxiul) [inline .travis.yml]: Should this be updated as well?

https://github.com/brave/browser-laptop/blob/836d3916f8941deca1aac65ec47aea66eae62b12/package.json#L230-L232

File: .travis.yml (hunk 1 above)
Comment (NejcZdovc) [inline .travis.yml]: :+1:

**Comment by NejcZdovc** [comment]
@ayumi are you working on this one right now? :smiley: @bsclifton said that you have some problems with travis, so I wanted to check it out and help out

**Comment by ayumi** [comment]
Currently blocked on fixing `--debug` on Linux:
https://github.com/brave/muon/issues/292

**Comment by luixxiul** [comment]
Is it not worth adding a large amount of cache to see how much it affects the performance?

**Comment by ayumi** [comment]
@luixxiul what do you mean?

this is no longer blocked on linux muon; i'm working around the `--debug` linux crash by using muon debug builds rather than using the prod builds.

**Comment by luixxiul** [comment]
I meant loading a cache file to calculate how much it can slow down the browser.

**Comment by cezaraugusto** [comment]
@ayumi is this ready for review? please put the `work-in-progress` label otherwise, thanks

**Comment by ayumi** [comment]
@cezaraugusto yes, this is ready for review

the tests have been reasonably deterministic.

**Comment by bsclifton** [comment]
@ayumi I'm trying to test this out and can use some help 😄 

1. Would you be able to create some docs to accompany this?  Maybe a new wiki entry:
https://github.com/brave/browser-laptop/wiki

2. I checked out this branch on an Ubuntu 14.04 VM, removed node_modules, and re-did the `npm install`.  After that completed, I ran the following:
```
curl -sL https://raw.githubusercontent.com/travis-ci/artifacts/master/install | bash
CXX=g++-4.8 NODE_ENV=test TEST_DIR=performance ARTIFACTS_REGION=us-east-1 npm run testsuite
```

When running this, I'm getting a few errors (some tests are timing out).  However, it all seems great on the Travis side (which is more important 😄 ):
https://travis-ci.org/brave/browser-laptop/jobs/286826649

Any advice on what to try the setup and test for?

3. You had shown me the Grafana instance (would be good to document some details about that).  This is uploading the results of a run to S3.  Once merged, this should be collecting all of the perf data about each PR, correct?

**Comment by bsclifton** [comment]
BTW- per discussion on Slack, the `--debug` option should work on Linux builds for the latest Muon.  If you get a chance, please retry that and let us know how it works (if resolved, we can close https://github.com/brave/muon/issues/292).  Until we get a chance to test that, we can still keep the separate debug version of Muon 😄

**Comment by ayumi** [comment]
1) :+1: 

2) travis artifacts uploader isn't needed outside of travis

3) Yes, after this is merged we will start collecting data on every Travis run for browser-laptop; and it will automatically go to the https://github.com/brave/perfaderp web UI.

4) `--debug` still doesn't work for me on Linux. browser-laptop master 3ee0c9543f582794117b3ce98832da339285429d

**Comment by bsclifton** [comment]
@ayumi the last commit has some failing tests.  I restarted the travis job- if that passes, this looks great; will merge 😄

**Comment by ayumi** [comment]
@bsclifton I might need to create a new muon debug build

**Comment by bsclifton** [comment]
@ayumi OK great, sounds good 😄 Let's merge this for now and you can investigate that 😄

**Comment by ayumi** [comment]
tests working again with a new muon debug build

**Pull request merged**